### ISSUE TITLE
Zoom or crop in Android app.

### ISSUE THREAD

**Issue by simonmikkelsen** [mapillary/mapillary_issues#1173]
Some phones have very vide lenses and when mapping from a car sequences like this one (of mine) is often seen: http://www.mapillary.com/map/im/fonrriN8PVRe-bVOjcwW7w
Part of the dash board in the bottom. Just pointing the camera more up just gives other problems (which I will file another issue on when I get the time).

I suggests that you can zoom or crop in the app to get better pics.

Perhaps both normal zoom or just "cut the bottom" zoom.

**Comment by domdomegg** [comment]
:+1:

Probably best to have the sensor at the highest resolution (no matter the aspect ratio) and have the user select a part to crop down to in the ratio 4:3, like most locked grid cropping systems

**Comment by simonmikkelsen** [comment]
I think that in many cases the most interesting part of an image is the sides. That is why I suggested a "cut the bottom". This will produce more like 16:9 ratio images and the system must handle these.

**Comment by domdomegg** [comment]
That does seem like a better solution, but I don't know if Mapillary can handle that. However, there has been a lot of pressure recently to switch from 4:3 to 16:9 due to (as you said) more useful information rather than car dashboards and clouds.

**Comment by simonmikkelsen** [comment]
I have uploaded about 100k photos in 16:9, about 16k in 4:3 and about 1k where I have taken the time to crop as I suggests here. Mappillary can handle it, but in the old/current viewer everything is just squashed into 4:3 format. I think the new one handles it better. Imho Mapillary should handle all formats within reasonable limitations, because they will get images in many different formats.

**Comment by flohoff** [comment]
Its not always cut the bottom. Currently when i shoot with a S4 i have the bonnet of the car on the lower part and in the right top corner i have parts of the cradle which holds the mobile. So i'd like to crop to whatever excludes my distortions in view. This can be everything in every part of the image. Whether its 16:9 or 4:3 doesnt matter - offer the user a method to crop to some 16:9 or 4:3 based area. I'd like to crop - currently i dont use this setup for shooting mapillary images for this reason.

**Comment by simonmikkelsen** [comment]
Yes, imho there are good use cases for both normal "digital zoom" and cropping any number of the sides.

**Comment by utack** [comment]
I thought this was kind of a given.  
This should be added quickly, or the database of poor quality pictures where dashboards are not cropped out will grow, at the moment there is perhaps a chance that every person could still manage to add this crop if it is also added in the webinterface

**Comment by peterneubauer** [comment]
While supporting zoom is one possible solution, it introduces errors for the focal length of the current shot which might not be as accurate as the default camera focal length.

We are thinking of instead on the server side and on device detect dashboards etc and either automatically zoom in the viewer to hide them, or make the user aware and get them to adjust the camera.

**Comment by utack** [comment]
So should I upload images with dashboard meanwhile or none until they are perfect?

**Comment by peterneubauer** [comment]
Yeah,
if it is not too much dashboard in the picture it's ok. Got a sample?

**Comment by utack** [comment]
Not really, until two days ago images were also rotated on my phone, so 
I have zero uploads until now


Am 01.05.2016 um 18:37 schrieb Peter Neubauer:
> Yeah,
> if it is not too much dashboard in the picture it's ok. Got a sample?
>

**Comment by domdomegg** [comment]
I just wanted to point out in some scenarios it is impractical to not get dashboard in without either being 3/4 sky or just a reflection of the dashboard anyway so just warning the user will not help.

**Comment by domdomegg** [comment]
(Or not help in all scenarios)

**Comment by GITNE** [comment]
> Probably best to have the sensor at the highest resolution (no matter the aspect ratio)

No, probably not. This is a very naive assumption, to say the least. Why? Because only in theory things work perfectly smooth. In realty, a plethora of devices, cameras, and software exist, including many permutations of those. In some devices cameras are backed by dedicated pre-processing hardware and hardware accelerated post-processing :smiley:, in others cameras simply deliver raw image data which has to be post-processed by a smartphone's CPU or GPU :disappointed:, which puts more stress on battery life or memory bandwidth. While even others have software controlled auto-focus and exposition sensors taking a significant amount of time and/or memory resources to adjust. Because of all this some devices can take full resolution photos fairly quickly others are quick capturing a photo but slow storing it to internal memory. No simple one size fits all solution can be applied here, regardless of how temping it may be to believe to have one.
And although I am sure you had good intentions, please abstain from making unsubstantiated quick shot assumptions while selling them as good advice.

> have the user select a part to crop down to in the ratio 4:3, like most locked grid cropping systems

This may not work well on all devices either, or as intended. That is, you will get 4:3 photos but with other massive side effects for the user which may even render the Mapillary app **unusable** on that particular device or family of devices. So, even though the app might run the side effects will quickly frustrate the user and make him or her give up.

What can you do? Two things: No 1, test, test, test, and once again test on different devices. Unfortunately, this can become quite expensive (lots of :heavy_dollar_sign::heavy_dollar_sign::heavy_dollar_sign:) even if you source this service and do not buy your own gear. No 2, put as many sane camera configuration options (variables) into the app as possible, preferably auto-configurable and auto-detected, perhaps even based on telemetry data as a default starting point. This default configuration can auto-adjust over time to the particular user's flow of use. This cyclic system approach is probably going to be cheaper in the long run but it takes some brains to build and time to setup and fine tune.
